util/timeutil/time.go

```diff
@@ -0,0 +1,161 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package timeutil
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// init initializes `locCache`.
+func init() {
+	// We need set systemTZ when it is in testing process.
+	if systemTZ == "" {
+		systemTZ = "System"
+	}
+	locCa = &locCache{}
+	locCa.locMap = make(map[string]*time.Location)
+}
+
+// locCa is a simple cache policy to improve the performance of 'time.LoadLocation'.
+var locCa *locCache
+
+// LocalStr is current TiDB's system timezone name.
+var systemTZ string
+var zoneSources = []string{
+	"/usr/share/zoneinfo/",
+	"/usr/share/lib/zoneinfo/",
+	"/usr/lib/locale/TZ/",
+	// this is for macOS
+	"/var/db/timezone/zoneinfo",
+}
+
+// locCache is a simple map with lock. It stores all used timezone during the lifetime of tidb instance.
+// Talked with Golang team about whether they can have some forms of cache policy available for programmer,
+// they suggests that only programmers knows which one is best for their use case.
+// For detail, please refer to: https://github.com/golang/go/issues/26106
+type locCache struct {
+	sync.RWMutex
+	// locMap stores locations used in past and can be retrieved by a timezone's name.
+	locMap map[string]*time.Location
+}
+
+// InferSystemTZ reads system timezone from `TZ`, the path of the soft link of `/etc/localtime`. If both of them are failed, system timezone will be set to `UTC`.
+// It is exported because we need to use it during bootstap stage. And it should be only used at that stage.
+func InferSystemTZ() string {
+	// consult $TZ to find the time zone to use.
+	// no $TZ means use the system default /etc/localtime.
+	// $TZ="" means use UTC.
+	// $TZ="foo" means use /usr/share/zoneinfo/foo.
+	tz, ok := syscall.Getenv("TZ")
+	switch {
+	case !ok:
+		path, err1 := filepath.EvalSymlinks("/etc/localtime")
+		if err1 == nil {
+			name, err2 := inferTZNameFromFileName(path)
+			if err2 == nil {
+				return name
+			}
+			log.Errorln(err2)
+		}
+		log.Errorln(err1)
+	case tz != "" && tz != "UTC":
+		for _, source := range zoneSources {
+			if _, err := os.Stat(source + tz); err == nil {
+				return tz
+			}
+		}
+	}
+	return "UTC"
+}
+
+// inferTZNameFromFileName gets IANA timezone name from zoneinfo path.
+// TODO: It will be refined later. This is just a quick fix.
+func inferTZNameFromFileName(path string) (string, error) {
+	// phase1 only support read /etc/localtime which is a softlink to zoneinfo file
+	substr := "zoneinfo"
+	if idx := strings.Index(path, substr); idx != -1 {
+		return string(path[idx+len(substr)+1:]), nil
+	}
+	return "", errors.New(fmt.Sprintf("path %s is not supported", path))
+}
+
+// Local returns time.Local's IANA timezone name. It is TiDB's global timezone name.
+func Local() *time.Location {
+	loc, err := LoadLocation(systemTZ)
```

Comment:
What happen when `Local()` is called before or during cluster bootstrapping ?
`systemTZ` is not initialized yet.